ddtrace/tracer/payload.go

```diff
@@ -108,9 +109,21 @@ func (p *payload) updateHeader() {
 func (p *payload) Read(b []byte) (n int, err error) {
 	if p.off < len(p.header) {
 		// reading header
-		n = copy(b, p.header[p.off:])
+		n, err := p.buf.Write(p.header[p.off:])
+		if err != nil {
+			return n, err
+		}
 		p.off += n
-		return n, nil
 	}
-	return p.buf.Read(b)
+
+	for len(p.traces) != 0 && p.buf.Len() <= len(b) {
```

Comment:
Let's put an inline comment inside the `for`, something like:
```go
// fill buffer
```

Reply:
Will do.